cider/tests/nextgen/translator/ColumnToRowTest.cpp

```diff
@@ -0,0 +1,137 @@
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "exec/module/batch/ArrowABI.h"
+#include "exec/nextgen/jitlib/base/ValueTypes.h"
+#include "exec/nextgen/operators/ColumnToRowNode.h"
+#include "exec/nextgen/operators/FilterNode.h"
+#include "exec/nextgen/operators/ProjectNode.h"
+#include "exec/nextgen/operators/SourceNode.h"
+#include "exec/plan/parser/TypeUtils.h"
+#include "tests/TestHelpers.h"
+#include "tests/nextgen/translator/MockRowToColumn.h"
+#include "tests/utils/ArrowArrayBuilder.h"
+
+using namespace cider::jitlib;
+using namespace cider::exec::nextgen;
+using namespace cider::exec::nextgen::operators;
+
+using ExprPtr = std::shared_ptr<Analyzer::Expr>;
+
+class ColumnToRowTest : public ::testing::Test {};
+
+ExprPtr makeConstant(int32_t val) {
+  Datum d;
+  d.intval = val;
+  return std::make_shared<Analyzer::Constant>(kINT, false, d);
+}
+
+TEST_F(ColumnToRowTest, FilterTest) {
+  LLVMJITModule module("TestC2R", true);
+
+  auto builder = [](JITFunction* function) {
+    //   if (col_var1 <= 5) {
+    //     res = col_var1 + 5;
+    //   }else{
+    //     return col_var1;
+
+    auto col_var1 =
+        std::make_shared<Analyzer::ColumnVar>(SQLTypeInfo(SQLTypes::kINT), 100, 1, 0);
+    auto col_var2 =
+        std::make_shared<Analyzer::ColumnVar>(SQLTypeInfo(SQLTypes::kDOUBLE), 100, 2, 0);
+
+    // constant
+    auto const_var = makeConstant(5);
+
+    // var <= 5
+    auto cmp_expr = std::make_shared<Analyzer::BinOper>(
+        SQLTypes::kBOOLEAN, SQLOps::kLE, SQLQualifier::kONE, col_var1, const_var);
+
+    // var + 5
+    auto add_expr = std::make_shared<Analyzer::BinOper>(
+        SQLTypes::kINT, SQLOps::kPLUS, SQLQualifier::kONE, col_var1, const_var);
+
+    // source -> column2row -> filter -> project -> sinkR2C
+    auto source = SourceTranslator(
+        std::vector<ExprPtr>{col_var1, col_var2},
+        std::make_unique<ColumnToRowTranslator>(
+            std::vector<ExprPtr>{col_var1, col_var2},
+            std::make_unique<FilterTranslator>(
+                cmp_expr,
+                std::make_unique<ProjectTranslator>(
+                    add_expr, std::make_unique<MockRowToColumnTranslator>(1)))));
```

Comment:
where is `return col_var1`?

Reply:
I construct MockRowToColumnTranslator class to return col_var1